src/frontend/admin/src/components/templates/courses/form/sections/product-relation/useCourseProductRelationList.tsx

```diff
@@ -0,0 +1,162 @@
+import { useMemo, useState } from "react";
+import { faker } from "@faker-js/faker";
+import { useModal } from "@/components/presentational/modal/useModal";
+import { useList } from "@/hooks/useList/useList";
+import {
+  CourseProductRelation,
+  CourseProductRelationDummy,
+  DTOCourseProductRelation,
+} from "@/services/api/models/Relations";
+import { CourseProductRelationFormValues } from "@/components/templates/courses/form/product-relation/CourseProductRelationForm";
+import { useCourseProductRelations } from "@/hooks/useCourseProductRelation/useCourseProductRelation";
+
+type EditRelationState = {
+  relation: CourseProductRelation;
+  index: number;
+};
+
+enum Mode {
+  EDIT = "edit",
+  ADD = "add",
+}
+
+type Props = {
+  courseId?: string;
+  productId?: string;
+  relations: CourseProductRelation[];
+  invalidate: () => void;
+};
+
+export const useCourseProductRelationList = ({
+  relations = [],
+  invalidate,
+}: Props) => {
+  const relationToProductModal = useModal();
+  const [relationToEdit, setRelationToEdit] = useState<EditRelationState>();
+  const modalForm = useModal();
+  const resourceQuery = useCourseProductRelations({}, { enabled: false });
+
+  const relationList = useList<CourseProductRelation>(relations);
+  const dummyRelationList = useList<CourseProductRelationDummy>([]);
+
+  const mode: Mode = useMemo(() => {
+    return relationToEdit?.relation !== undefined ? Mode.EDIT : Mode.ADD;
+  }, [relationToEdit?.relation]);
+
+  const handleCreate = (
+    payload: DTOCourseProductRelation,
+    formValues: CourseProductRelationFormValues,
+  ) => {
+    const dummy: CourseProductRelationDummy = {
+      ...formValues,
+      can_edit: false,
+      order_groups: [],
+      dummyId: faker.string.uuid(),
+    };
+
+    modalForm.handleClose();
+    dummyRelationList.clear();
+    dummyRelationList.push(dummy);
+    resourceQuery.methods.create(payload, {
+      onSuccess: (data) => {
+        relationList.insertAt(0, data);
+      },
+      onSettled: () => {
+        dummyRelationList.clear();
+      },
+    });
+  };
+
+  const afterEdit = (
+    editMode: "error" | "success",
+    data?: CourseProductRelation,
+  ) => {
+    if (!relationToEdit) {
+      return;
+    }
+
+    if (editMode === "error") {
+      relationList.updateAt(relationToEdit?.index, relationToEdit.relation);
+    }
+
+    if (editMode === "success" && data) {
+      relationList.updateAt(relationToEdit?.index, data);
+      setRelationToEdit(undefined);
+      resourceQuery.methods.invalidate();
+      invalidate();
+    }
```

Comment:
You are checking just before if `relationToEdit` is defined so no need to use `?`
```suggestion
    if (editMode === "error") {
      relationList.updateAt(relationToEdit.index, relationToEdit.relation);
    }

    if (editMode === "success" && data) {
      relationList.updateAt(relationToEdit.index, data);
      setRelationToEdit(undefined);
      resourceQuery.methods.invalidate();
      invalidate();
    }
```